File: src/main/java/com/example/thymleafexamples/dto/WorkOrderDTO.java
package com.example.thymleafexamples.dto;

import com.example.thymleafexamples.core.dto.BaseDTO;
import com.example.thymleafexamples.domain.enums.ReasonCategory;
import com.example.thymleafexamples.domain.enums.RecordType;
import com.example.thymleafexamples.domain.enums.SituationType;
import com.example.thymleafexamples.domain.enums.SolutionCategory;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;

@Data
public class WorkOrderDTO extends BaseDTO {
    private String opyCode;

    private String maximoId;

    private String serviceRegistrationNumber;
    private String assigned;
    private String reported;
    private String application;

    private String description;

    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
    private LocalDateTime completionDate;
    private RecordType recordType;
    private SituationType situationType;
    private ReasonCategory reasonCategory;
    private SolutionCategory solutionCategory;
}

File: fe-wop/src/types/dto/WorkOrderDTO.ts
export interface WorkOrderDTO {
    id?: string;
    opyCode: string;
    maximoId: string;
    serviceRegistrationNumber: string;
    assigned: string;
    reported: string;
    application: string;
    description: string;
    completionDate: string;
    recordType: RecordType;
    situationType: SituationType;
    reasonCategory: ReasonCategory;
    solutionCategory: SolutionCategory;
}

export enum RecordType {
    MAINTENANCE = "MAINTENANCE",
    INSPECTION = "INSPECTION",
    REPAIR = "REPAIR",
}

export enum ReasonCategory {
    EQUIPMENT_FAILURE = "EQUIPMENT_FAILURE",
    HUMAN_ERROR = "HUMAN_ERROR",
    EXTERNAL_FACTOR = "EXTERNAL_FACTOR",
}

export enum SituationType {
    OPEN = "OPEN",
    IN_PROGRESS = "IN_PROGRESS",
    CLOSED = "CLOSED",
}

export enum SolutionCategory {
    REPLACEMENT = "REPLACEMENT",
    REPAIR = "REPAIR",
    ADJUSTMENT = "ADJUSTMENT",
}